frontend/packages/dev-console/src/components/import/import-submit-utils.ts

```diff
@@ -250,28 +251,47 @@ export const createRoute = (
   return k8sCreate(RouteModel, route, dryRun ? dryRunOpt : {});
 };
 
-export const createResources = (
+export const createResources = async (
   formData: GitImportFormData,
   imageStream: K8sResourceKind,
   dryRun: boolean = false,
 ): Promise<K8sResourceKind[]> => {
   const {
+    application: { name: applicationName },
+    project: { name: projectName },
     route: { create: canCreateRoute },
     image: { ports },
   } = formData;
 
   const requests: Promise<K8sResourceKind>[] = [
-    createDeploymentConfig(formData, imageStream, dryRun),
     createImageStream(formData, imageStream, dryRun),
     createBuildConfig(formData, imageStream, dryRun),
   ];
 
-  if (!_.isEmpty(ports)) {
-    requests.push(createService(formData, imageStream, dryRun));
-    if (canCreateRoute) {
-      requests.push(createRoute(formData, imageStream, dryRun));
+  if (!formData.serverless.trigger) {
+    requests.push(createDeploymentConfig(formData, imageStream, dryRun));
+
+    if (!_.isEmpty(ports)) {
+      requests.push(createService(formData, imageStream, dryRun));
+      if (canCreateRoute) {
+        requests.push(createRoute(formData, imageStream, dryRun));
+      }
     }
+    return Promise.all(requests);
+  }
+
+  // knative service doesn't have dry run capability so returning the promises.
+  if (dryRun) {
+    return Promise.all(requests);
   }
 
-  return Promise.all(requests);
+  // invoke knative service if serverless option is selected
+  const [imageStreamResponse] = await Promise.all(requests);
+  return Promise.all([
```

Comment:
You don't need `Promise.all` here as there is only one promise to be resolved.

Reply:
I having difficulties changing the method signature. It is supposed to return a Promise array of type K8sResourceKind. When I tried to include a single promise, I'm getting type error. I may need some help here.